contributors/design-proposals/scheduling/scheduling-policy.md

```diff
@@ -0,0 +1,516 @@
+
+# Overview
+
+### syntaxic standards
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: <POLICY_KIND>
+metadata:
+  name: <POLICY_NAME>
+spec:
+  bindingMode: <BINDING_MODE>            # Describes a bindingMode (any or all)
+  namespaces:                            # a list of namespaces that the policy targets (optional)
+    - ns1
+    - ns1
+  namespaceSelector:                     # a selector to match a set of namespaces (optional)
+    key: value
+  rules:                                 # rules that must be satisfied (optional)
+    fieldA:                              # name of the field (optional)
+      - match:                             # Describes how the rule is matched (required)
+        - elt1                           # elements here could be objects like tolerations or strings like SchedulerName
+        action:                            # Describes the action that should be taken (allowed, denied or required)
+
+```
+
+### policy composition
+
+`bindingMode` Describes how policies should be composed:
+
+- any: any policy that has its rules satisfied
+- all: all policies MUST have its rules satisfied
+
+if we have a heterogenous bindingMode across policies (i.e. some policies with any and others with all).
+Then the most restrictive one (i.e the all bindingMode) is applied.
+
+### unset fields:
+
+if a field was not specified in the policy no rule should apply (i.e. everything is allowed).
+
+### empty match:
+
+Usually policies should distinguish between empty matches, since it depends on the action:
+
+- require/deny: no rule apply.
+- allow: allows everything.
+
+
+### inside the matches:
+
+The match field can express pretty much any structure you want to, but there's some things that should be considered:
+
+- the structure should match as much as possible what you try to take action on.
+- match elements are ANDed.
+- the `*` wildcard is usually needed in the allow section. It should be represented with an empty value (e.g. empty array).
+
+
+### conflict handling for policies
+
+There is two kinds of conflict in policies:
+
+- structural conflicts: these exists when any rule has the same matchingExpression and opposed actions (deny and require)
+
+- semantical conflicts: these exists due to semantic of the fields (e.g. require a NodeSelector `master=true` and also require `master=false`)
+
+
+Structural conflicts must handled at creation time, on the other hand, semantical conflicts should be handled at runtime: detect that there's a conflict and emit an event stating that policies couldn't be satisfied due to a conflict.
+
+
+### required vs allowed vs denied
+
+Policies may have overlapping rules, to handle this policies are computed in the following order:
+
+- compute what is denied.
+- compute what is required.
+- compute what was allowed.
+
+they should also obey to the following rules:
+
+- everything that is required is by definition allowed.
+- everything that is not denied is not automatically allowed.
```

Comment:
This means that the policy is "default deny". We cannot easily roll-out "default deny" policies in existing clusters.

Reply:
see @liggitt's comment and mine.